android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/common/notification/NotificationChannelType.kt

```diff
@@ -0,0 +1,41 @@
+package com.on.staccato.presentation.common.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+
+enum class NotificationChannelType(
+    val id: String,
+    val importance: Int,
+) {
+    Invitation(
+        id = "invitation",
+        importance = NotificationManager.IMPORTANCE_HIGH,
+    ),
+    Category(
+        id = "category",
+        importance = NotificationManager.IMPORTANCE_HIGH,
+    ),
+    Staccato(
+        id = "staccato",
+        importance = NotificationManager.IMPORTANCE_HIGH,
+    ),
+    Comment(
+        id = "comment",
+        importance = NotificationManager.IMPORTANCE_HIGH,
+    ),
+    ;
+
+    companion object {
+        private fun NotificationChannelType.toNotificationChannel() = NotificationChannel(id, name, importance)
+
+        fun toChannel(title: String): NotificationChannel {
+            return when {
+                title.contains("님이 초대를 보냈어요") -> Invitation.toNotificationChannel()
+                title.contains("님이 참여했어요") -> Category.toNotificationChannel()
+                title.contains("스타카토가 추가됐어요") -> Staccato.toNotificationChannel()
+                title.contains("님의 코멘트") -> Comment.toNotificationChannel()
+                else -> Staccato.toNotificationChannel()
+            }
+        }
```

Comment:
onMessageReceived에서 받아온 RemoteMessage title의 suffix를 활용해 타입을 구분해 주셨군요! 아이디어 짱이햐 👍
title의 suffix가 알림의 타입을 구분하는 중요한 역할을 하고 있네요! 

그렇다면 각 알림의 타입이 suffix를 가지고 있고, 이를 기반으로 아래와 같이 channel로 변환하는 방식은 어떨까요?

```suggestion
        fun toChannel(title: String): NotificationChannel = fromTitle(title).toNotificationChannel()

        private fun fromTitle(title: String): NotificationChannelType = entries.firstOrNull { title.contains(it.suffix) } ?: Staccato
```

```kotlin
 enum class NotificationChannelType(
    val id: String,
    val suffix: String,
    val importance: Int,
) {
    Invitation(
        id = "invitation",
        suffix = "님이 초대를 보냈어요",
        importance = NotificationManager.IMPORTANCE_HIGH,
    ),
    ...
}
```

Reply:
으아앗 suffix를 활용하는 부분은 임시 구현이라, Data payload로 변경되면 사라질 부분입니다 하핫..!